src/gcp_scanner/scanner.py

```diff
@@ -269,12 +271,20 @@ def crawl_loop(initial_sa_tuples: List[Tuple[str, Credentials, List[str]]],
 
       sa_results_data = json.dumps(sa_results, indent=2, sort_keys=False)
 
-      with open(out_dir + '/%s.json' % project_id, 'a',
-                encoding='utf-8') as outfile:
-        outfile.write(sa_results_data)
-
-      # Clean memory to avoid leak for large amount projects.
-      sa_results.clear()
+      # Generate current timestamp to append to output filename
+      scan_time_suffix = datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
+      output_file_name = f'{project_id}-{scan_time_suffix}.json'
+      output_path = Path(out_dir, output_file_name)
+      
+      try:
+        with open(output_path, 'x', encoding='utf-8') as outfile:
+          outfile.write(sa_results_data)
+
+        # Clean memory to avoid leak for large amount projects.
+        sa_results.clear()
+      except FileExistsError:
```

Comment:
Please do not fail here. It can be very time expensive to wait until here to know that file already exist. Show a warning and replace/append file if that's the case. Additionally, it is quite unlikely to happen given you already have timestamp...

Reply:
Probably not a good idea to replace/append. You can just append _2 or something to file name if it is a duplicate.

Reply:
@ZetaTwo that would be even better.

Reply:
Yep, agree.

Reply:
Implemented the suggestions :)